log/format.go

```diff
@@ -431,31 +434,32 @@ func formatLogfmtUint64(n uint64, neg bool) string {
 	return string(out[i+1:])
 }
 
-var big1000 = big.NewInt(1000)
-
-// formatLogfmtBigInt formats a potentially gigantic number in a friendlier split
-// format.
+// formatLogfmtBigInt formats n with thousand separators.
 func formatLogfmtBigInt(n *big.Int) string {
-	// Most number don't need fancy handling, just downcast
 	if n.IsUint64() {
 		return FormatLogfmtUint64(n.Uint64())
 	}
 	if n.IsInt64() {
 		return FormatLogfmtInt64(n.Int64())
 	}
-	// Ok, huge number needs huge effort
-	groups := make([]string, 0, 8) // random initial size to cover most cases
-	for n.Cmp(big1000) >= 0 {
-		_, mod := n.DivMod(n, big1000, nil)
-		groups = append(groups, fmt.Sprintf("%03d", mod))
-	}
-	groups = append(groups, n.String())
 
-	last := len(groups) - 1
-	for i := 0; i < len(groups)/2; i++ {
-		groups[i], groups[last-i] = groups[last-i], groups[i]
+	text := n.String()
+	buf := make([]byte, 0, len(text)+len(text)/3)
+	comma := 0
+	for _, c := range text {
```

Comment:
You need to construct it backwards, otherwise you don't know where the commas need to be

```
=== RUN   TestPrettyBigInt
    format_test.go:76: invalid output 111,222,333,444,555,678,999,00, want 11,122,233,344,455,567,899,900
    format_test.go:76: invalid output -111,222,333,444,555,678,999,00, want -11,122,233,344,455,567,899,900
```